### PR TITLE
Pick up new CoreFX packages and abandon old ones

### PR DESCRIPTION
**Pull request by wtgodbe**
This change does a few things:

- Separates our external dependencies into external/project.json
- Changes all dependencies on System.xyz to Microsoft.Private.CoreFx.NETCoreApp
- Removes several JIT/config project.json files (Everything except benchmark, benchmark+roslyn, benchmark+serialize), and re-assigns all test projects who previously depended on those, to depend on test-dependencies/project.json
- Adds external.depproj, which handles restoring dependencies from external/project.json into a Targeting Pack folder, which tests reference directly.
- Moves from netcoreapp1.1 to netcoreapp2.0, where possible.

There's a lot of changes here, but the relevant files should be at the top of the changeset - everything else is either a deletion of a project.json file, or a deletion of a Property referencing one of those files. (Or @AndyAyersMS fixing a bad ReferencePath in an .ilproj).

I did not touch any of the Perf stuff, as those tests still depend on using NetStandard1.4. Once that changes, we should be able to get rid of those project.json's as well, and point perf tests at test-dependencies/project.json.

### CODE REVIEW

**Comment by wtgodbe** [comment]
CC @gkhanna79 @jkotas @ramarag @mellinoe @AndyAyersMS @DrewScoggins @weshaggard @ericstj

**Comment by wtgodbe** [comment]
CC @pgavlin as well, for the JIT stuff

**Comment by gkhanna79** [comment]
Is the Pri1 test build and run clean with it locally and (clean run) in Helix?

**Comment by wtgodbe** [comment]
@gkhanna79 I sent a few jobs to helix that came back clean, but haven't done a full pri-1 job yet - I'm setting that up now

File: tests/src/Common/test_dependencies/project.json [hunk 1]
```diff
@@ -1,79 +1,15 @@
...
-    "xunit.console.netcore": "1.0.2-prerelease-00177",
-    "xunit.runner.utility": "2.2.0-beta2-build3300",
-    "Microsoft.CodeAnalysis.Compilers": "1.1.1",
-    "Microsoft.DotNet.xunit.performance": "1.0.0-alpha-build0040",
-    "Microsoft.DotNet.xunit.performance.analysis": "1.0.0-alpha-build0040",
-    "Microsoft.DotNet.xunit.performance.runner.Windows": "1.0.0-alpha-build0040",
-    "Microsoft.Win32.Primitives": "4.4.0-beta-24913-02",
-    "Newtonsoft.Json": "8.0.3",
     "Microsoft.NETCore.Platforms": "2.0.0-beta-25109-03",
     "Microsoft.NETCore.Targets": "1.2.0-beta-24913-02",
```
Comment (ericstj) [inline tests/src/Common/test_dependencies/project.json]: This package is not needed.

File: tests/src/Common/test_runtime/project.json [hunk 2]
```diff
@@ -7,9 +7,10 @@
     "jit-analyze": "0.0.1.1"
   },
   "frameworks": {
-    "netcoreapp1.1": {
+    "netcoreapp2.0": {
       "imports": [
         "dnxcore50",
+        "netcoreapp1.1",
```
Comment (ericstj) [inline tests/src/Common/test_runtime/project.json]: This is not needed.

**Comment by ericstj** [comment]
You may need some additional package refs.  We still produce packages for those listed here: https://github.com/dotnet/versions/blob/master/build-info/dotnet/corefx/master/Latest_Packages.txt.  Some of those overlap with NETCore.App but there is more than just S.R.CS.Unsafe that does not.

**Comment by mellinoe** [comment]
LGTM. We can add the extra package references now, or later when folks start wanting to use them. As long as the current suite of tests is satisfied then that seems ok to me.

**Comment by wtgodbe** [comment]
@gkhanna79 @weshaggard @ericstj has the UnicodeEncoding class changed recently? Looks like https://github.com/dotnet/coreclr/blob/master/tests/src/CoreMangLib/cti/system/text/unicodeencoding/unicodeencodinggetbytecount1.cs#L192 may need to change, based on the current errors.

Also, with the changes from https://github.com/dotnet/coreclr/pull/10150, tests are passing in Helix.

**Comment by wtgodbe** [comment]
@dotnet-bot test OSX x64 Checked Build and Test

**Comment by jkotas** [comment]
I have changed it in https://github.com/dotnet/coreclr/pull/10124 . The CI was green with my change. Is this running more tests?

**Comment by jkotas** [comment]
> https://github.com/dotnet/coreclr/blob/master/tests/src/CoreMangLib/cti/system/text/unicodeencoding/unicodeencodinggetbytecount1.cs#L192 

Yes, it needs to change. It is build break caused by adding a new API. It is expected - it was discussed during the review and determined to be ok.

**Comment by wtgodbe** [comment]
@jkotas what should it be changed to?

**Comment by jkotas** [comment]
`(char[])null, 0, 0`. Or delete the test - there is redundant test in corefx with much more comprehensive coverage.

**Comment by wtgodbe** [comment]
I'll just insert the fix for now.

**Comment by AndyAyersMS** [comment]
Fixes for the odd path corruptions I introduced look good (at least a sampling of them). Thanks for fixing this.

**Pull request merged**